### PR TITLE
fix!: Update alert appearance, make ‘warning’ the default and icon non-optional

### PR DESCRIPTION
**Pull request by VincentSmedinga**
- Updates to latest design: background now white, corrected text levels
- Adds a border token; not sure if the current set is exactly correct
- Uses `Heading` component to display title
- Adds ‘warning’ severity and makes it default
- Removes `icon` prop – distinguishing colour alone is not enough
- Hmm, this could / should have been 3 separate PRs
- Extends and organizes documentation, although not complete yet

### CODE REVIEW

File: packages/css/src/components/alert/alert.scss [hunk 1]
```diff
@@ -21,38 +21,25 @@
...
 }
 
+.amsterdam-alert--warning {
+  border-color: var(--amsterdam-alert-warning-border-color);
+}
+
 /* todo: move to action button component */
 .amsterdam-alert__close {
-  background-color: var(--amsterdam-alert-close-background-color);
+  background-color: var(--amsterdam-alert-close-button-background-color);
```
Comment (dlnr) [inline packages/css/src/components/alert/alert.scss:42]: This is a wasted effort, we should wait for the IconButton PR and then use that component. @alimpens has left out Alert on purpose.

File: packages/react/src/Alert/Alert.test.tsx [hunk 2]
```diff
@@ -8,9 +8,11 @@ describe('Alert', () => {
     const { container } = render(<Alert />)
 
     const component = container.querySelector(':only-child')
+    const icon = component?.querySelector('.amsterdam-alert__icon')
```
Comment (dlnr) [inline packages/react/src/Alert/Alert.test.tsx:11]: Whenever possible use `screen.getByRole()`. 

Side note: Perhaps also a good reason the think about using `role="alert"` for the error severity.

File: proprietary/tokens/src/components/amsterdam/alert.tokens.json [hunk 3]
```diff
@@ -1,36 +1,42 @@
 {
   "amsterdam": {
     "alert": {
-      "background-color": { "value": "{amsterdam.color.yellow}" },
-      "border": { "value": "4px solid {amsterdam.color.yellow}" },
+      "border-width": { "value": "{amsterdam.border-width.xl}" },
+      "border-style": { "value": "solid" },
```
Comment (dlnr) [inline proprietary/tokens/src/components/amsterdam/alert.tokens.json:5]: Not added to common border tokens?

File: proprietary/tokens/src/components/amsterdam/alert.tokens.json [hunk 4]
```diff
@@ -1,36 +1,42 @@
...
       "gap": { "value": "1rem" },
       "padding-block-start": { "value": "{amsterdam.spacing.inset.md}" },
       "padding-block-end": { "value": "{amsterdam.spacing.inset.md}" },
       "padding-inline-start": { "value": "{amsterdam.spacing.inset.lg}" },
       "padding-inline-end": { "value": "{amsterdam.spacing.inset.lg}" },
+      "close-button": {
```
Comment (dlnr) [inline proprietary/tokens/src/components/amsterdam/alert.tokens.json:11]: This is also not needed, we need to check why actionButton is applied to everything except Alert in Figma.

File: proprietary/tokens/src/components/amsterdam/alert.tokens.json (hunk 3 above)
Comment (VincentSmedinga) [reply proprietary/tokens/src/components/amsterdam/alert.tokens.json:5]: Yes it was, in a69bb9acd95fce6ef0a786d614b8f319a34ba34c.

File: proprietary/tokens/src/components/amsterdam/alert.tokens.json (hunk 4 above)
Comment (VincentSmedinga) [reply proprietary/tokens/src/components/amsterdam/alert.tokens.json:11]: It is in this PR, we’ll remove it again when applying Icon Button.

File: storybook/storybook-react/src/Alert/Alert.docs.mdx [hunk 5]
```diff
@@ -10,18 +10,46 @@ import README from "../../../../packages/css/src/components/alert/README.md?raw"
...
+### Success
+
+Laat het weten als een belangrijke actie geslaagd is.
+
+<Canvas of={AlertStories.Success} />
+
+### Warning
 
-<Canvas of={AlertStories.InformationWithInlineLink} />
+Toon een waarschuwing als er actie van de Amsterdammer nodig is.
```
Comment (alimpens) [inline storybook/storybook-react/src/Alert/Alert.docs.mdx:21]: This is a bit specific, and maybe not always correct?

```suggestion
Toon een waarschuwing als er actie van de gebruiker nodig is.
```

File: storybook/storybook-react/src/Alert/Alert.stories.tsx [hunk 6]
```diff
@@ -12,28 +12,21 @@ const meta = {
...
     title: 'Let op',
     closeable: false,
-    icon: false,
   },
   argTypes: {
     severity: {
       control: {
         type: 'radio',
-        labels: { undefined: 'default', error: 'error', success: 'success' },
+        labels: { warning: 'warning', success: 'success', error: 'error', info: 'info' },
```
Comment (alimpens) [inline storybook/storybook-react/src/Alert/Alert.stories.tsx:20]: I don't think the labels are necessary if they're the same as the options

File: storybook/storybook-react/src/Alert/Alert.docs.mdx (hunk 5 above)
Comment (VincentSmedinga) [reply storybook/storybook-react/src/Alert/Alert.docs.mdx:21]: We’re using this in multiple places already. I think it’s a guideline, but the Schrijfwijzer is currently offline. Even though it’s not always technically correct, it’s more personal than ‘user’. We’ll be working with a microscopy specialist soon, let’s revisit later.

File: storybook/storybook-react/src/Alert/Alert.stories.tsx (hunk 6 above)
Comment (VincentSmedinga) [reply storybook/storybook-react/src/Alert/Alert.stories.tsx:20]: Correct

File: storybook/storybook-react/src/Alert/Alert.docs.mdx (hunk 5 above)
Comment (alimpens) [reply storybook/storybook-react/src/Alert/Alert.docs.mdx:19]: We use 'gebruiker' more often than 'Amsterdammer' though, and both aren't very personal imo. I'd use 'gebruiker' for now, and maybe change it later?

File: packages/react/src/Alert/Alert.tsx [hunk 7]
```diff
@@ -3,22 +3,33 @@
...
 
 export interface AlertProps extends PropsWithChildren<HTMLAttributes<HTMLDivElement>> {
-  title?: string
-  severity?: undefined | 'error' | 'success'
+  /** Whether the alert can be dismissed by the user. Adds a button to the top right. */
   closeable?: boolean
-  icon?: boolean
+  /**
+   * The hierarchical level of the alert title within the document.
+   * @default 4
```
Comment (alimpens) [inline packages/react/src/Alert/Alert.tsx:20]:
```suggestion
   * @default 2
```

File: proprietary/tokens/src/components/amsterdam/alert.tokens.json (hunk 3 above)
Comment (VincentSmedinga) [reply proprietary/tokens/src/components/amsterdam/alert.tokens.json:5]: And in the meantime we’re no longer using this token.

File: packages/react/src/Alert/Alert.tsx (hunk 7 above)
Comment (VincentSmedinga) [reply packages/react/src/Alert/Alert.tsx]: Sharp as a knife, you :)

I’ll add a task to decide explicitly how far we want to go with JSDoc comments.

File: storybook/storybook-react/src/Alert/Alert.docs.mdx (hunk 5 above)
Comment (VincentSmedinga) [reply storybook/storybook-react/src/Alert/Alert.docs.mdx]: Alright, changed everything to ‘gebruiker’ for now.

**Review by alimpens — APPROVED** [review]

**Pull request merged**